File: Programming-Basics-with-Python/6.Conditional Statements Advanced - Exercise/04_fishing_boat.py
group_budget = int(input())
season = input()
fishermans = int(input())

spring_price = 3000
summer_price = 4200
autumn_price = 4200
winter_price = 2600

additional_discount = 0.00
trip_price = 0.00

if fishermans <= 6:
    additional_discount += 0.10
elif 7 <= fishermans <= 11:
    additional_discount += 0.15
elif fishermans >= 12:
    additional_discount += 0.25

if season == 'Summer' or season == 'Autumn':
    trip_price = summer_price - (summer_price * additional_discount)
elif season == 'Spring':
    trip_price = spring_price - (spring_price * additional_discount)
elif season == 'Winter':
    trip_price = winter_price - (winter_price * additional_discount)

if season != 'Autumn' and fishermans % 2 == 0:
    trip_price -= trip_price * 0.05

if group_budget >= trip_price:
    print(f'Yes! You have {group_budget - trip_price:.2f} leva left.')
else:
    print(f'Not enough money! You need {trip_price - group_budget:.2f} leva.')

File: Programming-Basics-with-JavaScript/3.Conditional Statements Advanced/04fishingBoat.js
function fishingBoat(input){
    let budget = Number(input[0])
    let season = String(input[1])
    let numberFishermen = Number(input[2])
    let rent = 0
    let extraDiscount = 0
    let discount = 0
  
    //  "Spring", "Summer", "Autumn", "Winter"
    if (season == 'Spring'){
        rent = 3000
        if (numberFishermen % 2 == 0){
            extraDiscount = rent * 0.05
        }if (numberFishermen <= 6) {
            discount = rent * 0.10
        }else if (numberFishermen > 7 && numberFishermen <= 11) {
            discount = rent * 0.15
        }else if (numberFishermen > 12) {
            discount = rent * 0.25
        }
    }else if (season == "Summer"){
        rent = 4200
        if (numberFishermen % 2 == 0){
            extraDiscount = rent * 0.05
        }if (numberFishermen <= 6) {
            discount = rent * 0.10
        }else if (numberFishermen > 7 && numberFishermen <= 11) {
            discount = rent * 0.15
        }else if (numberFishermen > 12) {
            discount = rent * 0.25
        }
    }else if (season == 'Autumn'){
        rent = 4200
        if (numberFishermen <= 6) {
            discount = rent * 0.10
        }else if (numberFishermen > 7 && numberFishermen <= 11) {
            discount = rent * 0.15
        }else if (numberFishermen > 12) {
            discount = rent * 0.25
        }
    }else if (season == 'Winter'){
        rent = 2600
        if (numberFishermen % 2 == 0){
            extraDiscount = rent * 0.05
        }if (numberFishermen <= 6) {
            discount = rent * 0.10
        }else if (numberFishermen > 7 && numberFishermen <= 11) {
            discount = rent * 0.15
        }else if (numberFishermen > 12) {
            discount = rent * 0.25
        }
    }
    let result = budget - (rent - extraDiscount - discount)
    if (result < 0){
        console.log(`Not enough money! You need ${Math.abs(result).toFixed(2)} leva.`)
    }else {
        console.log(`Yes! You have ${Math.abs(result).toFixed(2)} leva left.`)
    }
}
fishingBoat(["2000",
"Winter",
"13"])

